File: DarkOrbit 10.0/Net/netty/commands/DroneFormationChangeCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class DroneFormationChangeCommand
    {
        public static short ID = 4159;

        public static byte[] write(int uid, int selectedFormationId)
        {
            ByteArray param1 = new ByteArray(ID);
            param1.writeShort(-29097);
            param1.writeShort(28868);
            param1.writeInt(uid << 2 | uid >> 30);
            param1.writeInt(selectedFormationId >> 10 | selectedFormationId << 22);
            return param1.ToByteArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/DroneFormationChangeCommand.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

import simulator.netty.ServerCommand;

public class DroneFormationChangeCommand
        implements ServerCommand {

    public DroneFormationChangeCommand(int param1, int param2) {
        this.uid = param1;
        this.selectedFormationId = param2;
    }

    public static int ID = 8735;

    public int selectedFormationId = 0;
    public int uid                 = 0;

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeShort(-29201);
            param1.writeInt(this.selectedFormationId >>> 12 | this.selectedFormationId << 20);
            param1.writeInt(this.uid << 11 | this.uid >>> 21);
            param1.writeShort(29445);
        } catch (IOException e) {
        }
    }
}